src/modules/apps/link.ts

```diff
@@ -257,19 +253,38 @@ const performInitialLink = async (appId: string, builder: Builder, extraData : {
   log.debug('Sending files')
   filesWithContent.forEach(p => log.debug(p.path))
 
-  try {
-    const { code } = await builder.linkApp(appId, filesWithContent, linkOptions)
-    if (code !== 'build.accepted') {
-      throw new Error('Please, update your builder-hub to the latest version!')
+  const retryOpts = {
+    retries: 2,
+    minTimeout: 1000,
+    factor: 2,
+  }
+
+  const linkApp = async (bail: any, tryCount: number) => {
+    // wrapper for builder.linkApp to be used with the retry function below.
+    if (tryCount > 1) {
+      log.info(`Retrying...${tryCount-1}`)
     }
-  } catch (e) {
-    const data = e.response && e.response.data
-    if (data && data.code && data.code === 'build_in_progress') {
-      log.warn(`Build for ${appId} is already in progress`)
-    } else {
-      throw e
+    let linkOptions
+    const stickyHint = await getMostAvailableHost(appId, builder, N_HOSTS, AVAILABILITY_TIMEOUT)
+    linkOptions = { sticky: true, stickyHint }
+    try {
+      const { code } = await builder.linkApp(appId, filesWithContent, linkOptions)
+      if (code !== 'build.accepted') {
+        bail(new Error('Please, update your builder-hub to the latest version!'))
+      }
+    } catch (err) {
+      const data = err.response && err.response.data
+      if (data && data.code && data.code === 'build_in_progress') {
+        log.warn(`Build for ${appId} is already in progress`)
+        bail()
+      }
+      const statusMessage = err.response.status ?
+        `: Status ${err.response.status}` : null
+      log.error(`Error linking app${statusMessage}`)
```

Comment:
nitpicking: this can log `Error linking appnull`, right?

Reply:
Changed to empty string.